app/src/main/java/com/mk/tasky/authentication/login/presentation/LoginScreen.kt

```diff
@@ -16,16 +17,26 @@ import com.mk.tasky.R
 import com.mk.tasky.core.presentation.TaskyButton
 import com.mk.tasky.core.presentation.TaskyEmailTextField
 import com.mk.tasky.core.presentation.TaskyPasswordTextField
+import com.mk.tasky.core.util.UIEvent
 import com.mk.tasky.ui.theme.Gray
 import com.mk.tasky.ui.theme.Link
 
 @Composable
 fun LoginScreen(
     signupClick: () -> Unit,
+    onLogin: () -> Unit,
     viewModel: LoginViewModel = hiltViewModel()
 ) {
     val state = viewModel.state
 
+    LaunchedEffect(key1 = true) {
+        viewModel.uiEvent.collect {
+            if (it is UIEvent.Navigate) {
+                onLogin()
+            }
```

Comment:
Around the end of yesterdays meeting (recording is online), I talked about how you could model such events as state. I personally stopped using such event channels/flows after an interesting talk I saw at Droidcon. I did it like this here for a long time as well (and the UIEvent class also feels very familiar :D ), but now I feel like that doesn't fit that well into a fully state-driven environment like Compose.

But that's just my view, if you like using such events, there's nothing wrong about it as long as you correctly collect them. Just wanted throw that in :)

Reply:
Will check the Q&A's recording to learn about it. Thanks!